code/game/turfs/simulated/water.dm

```diff
@@ -86,20 +86,27 @@
 	return return_air() // Otherwise their head is above the water, so get the air from the atmosphere instead.
 
 /turf/simulated/floor/beach/water/Entered(atom/movable/AM, atom/oldloc)
+	wash_floor(get_turf(src))
+	START_PROCESSING(SSprocessing, src)
```

Comment:
There's another thing I noticed to, you're starting processing on the tile every time an object is entered, and stopping it every time an object leaves. The problem with this is that if two objects enter the pool tile, it will call start processing twice which is fine because it handles repetitions, however if you leave the pool, it will call stop processing and it will tell the processing to stop. There is no processing stacking.

Think of it like this. Two men enter a store. The store activates service when each man comes in. One of the men leaves, it stops service, and the man who is inside is no longer getting service.